src/Lecture2.hs

```diff
@@ -70,7 +79,52 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+
+-- My first intuition was to use fold. But using foldl' left me with a list reversed,
+-- and using foldr while giving result list in order was counting index from the
```

Comment:
Not everything needs to be written with fold 🙂 
Explicit recursion, pattern matching and using other functions are better usually 👍🏻